### PR TITLE
HDDS-2566. Handle InterruptedException in ContainerDataScanner

### PR DESCRIPTION
**Pull request by dineshchitlangia**
## What changes were proposed in this pull request?

Handled InterruptedException and addressed other Sonar code smells.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2566

## How was this patch tested?

Clean build, checkstyle, sonar in local.

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java [hunk 1]
```diff
@@ -49,6 +49,7 @@
   private final Canceler canceler;
   private final ContainerDataScrubberMetrics metrics;
   private final long dataScanInterval;
+  private static final String NAME = "ContainerDataScanner(";
```
Comment (vivekratnavel) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java:52]: Can we use String format here?
```suggestion
  private static final String NAME_FORMAT = "ContainerDataScanner(%s)";
```

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java [hunk 2]
```diff
@@ -66,7 +67,7 @@ public ContainerDataScanner(ContainerScrubberConfiguration conf,
     throttler = new HddsDataTransferThrottler(conf.getBandwidthPerVolume());
     canceler = new Canceler();
     metrics = ContainerDataScrubberMetrics.create(volume.toString());
-    setName("ContainerDataScanner(" + volume + ")");
+    setName(NAME + volume + ")");
```
Comment (vivekratnavel) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java:70]:
```suggestion
    setName(String.format(NAME_FORMAT, volume));
```

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java [hunk 3]
```diff
@@ -162,13 +165,14 @@ private static void logScanCompleted(
 
   public synchronized void shutdown() {
     this.stopping = true;
-    this.canceler.cancel("ContainerDataScanner("+volume+") is shutting down");
+    this.canceler.cancel(NAME + volume + ") is shutting down");
```
Comment (vivekratnavel) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java:168]:
```suggestion
    this.canceler.cancel(String.format(NAME_FORMAT, volume) + " is shutting down");
```

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java [hunk 4]
```diff
@@ -179,8 +183,7 @@ public ContainerDataScrubberMetrics getMetrics() {
 
   @Override
   public String toString() {
-    return "ContainerDataScanner(" + volume +
-        ", " + volume.getStorageID() + ")";
+    return NAME + volume + ", " + volume.getStorageID() + ")";
```
Comment (vivekratnavel) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java:186]:
```suggestion
    return String.format(NAME_FORMAT, volume + ", " + volume.getStorageID());
```

**Comment by dineshchitlangia** [comment]
@vivekratnavel thanks for review and suggestions. As all suggestions are incorporated and we got green CI, I am merging this.

**Pull request merged**